### PR TITLE
Append a table in bigquery

### PR DESCRIPTION
**Pull request by abronte**
This allows appending rows to a table by overriding the append method in a task.

For example:

```
class MyQueryTableStep1(luigi_bigquery.QueryTable):
    def append(self):
        return True

    def dataset(self):
        return 'luigi'

    def table(self):
        return 'tmp'

    def query(self):
        return "SELECT count(*) cnt FROM [publicdata:samples.github_nested]"
```

I'm sort of new to luigi and not really a python dev, so if theres a better way to accomplish this let me know!

### CODE REVIEW

File: luigi_bigquery/task.py [hunk 1]
```diff
@@ -119,7 +119,7 @@ def requires(self):
         return DatasetTask(self.dataset())
 
     def output(self):
-        return TableTarget(self.dataset(), self.table())
+        return TableTarget(self.dataset(), self.table(), None, False, None, self.append())
```
Comment (hakobera) [inline luigi_bigquery/task.py]: @abronte I think it's not necessary to rewrite `self.write_disposition`.
Change code to

``` python
    def _append(self):
        return self.write_disposition == bigquery.JOB_WRITE_APPEND

    def output(self):
        return TableTarget(self.dataset(), self.table(), append=self._append())
```

is enough. If you do so, just use `QueryTable` like this.

```
class MyQueryTableStep1(luigi_bigquery.QueryTable):
    write_disposition == bigquery.JOB_WRITE_APPEND

    def dataset(self):
        return 'luigi'

    def table(self):
        return 'tmp'

    def query(self):
        return "SELECT count(*) cnt FROM [publicdata:samples.github_nested]"
```

**Comment by abronte** [comment]
Thanks! I updated the code so you can append a table just be setting `write_disposition` in a task.

Your example should work now:

```
class MyQueryTableStep1(luigi_bigquery.QueryTable):
    write_disposition = bigquery.JOB_WRITE_APPEND

    def dataset(self):
        return 'luigi'

    def table(self):
        return 'tmp'

    def query(self):
        return "SELECT count(*) cnt FROM [publicdata:samples.github_nested]"
```

File: luigi_bigquery/task.py [hunk 2]
```diff
@@ -134,10 +134,16 @@ def dataset(self):
...
 
+    def _append(self):
+        return self.write_disposition == bigquery.JOB_WRITE_APPEND
+
     def save_as_table(self, query):
         result = self.output()
         client = self.config.get_client()
 
+        if self.append():
+            self.write_disposition = bigquery.JOB_WRITE_APPEND
```
Comment (hakobera) [inline luigi_bigquery/task.py]: Please delete above 2 lines. It is not necessary now.

**Comment by abronte** [comment]
I removed those two lines now. Thanks!

**Comment by hakobera** [comment]
@abronte Thanks!

**Pull request merged**